common/usr/share/MailScanner/perl/MailScanner/MSMail.pm

```diff
@@ -1089,7 +1089,13 @@ sub new {
                      s/^\<//;
                      s/\>$//;
                   }
-                  my $payload = join '', map {sprintf "%d:%s,", length $_, $_} <$queuehandle>, $sender, @recipient;
+                  # The queue file contains <CRLF> line terminators in the
+                  # message body instead of <LF>. SMTP delivery handles these,
+                  # but QMQP keeps them in the delivered message which breaks
+                  # signatures.
+                  my $messagetext = <$queuehandle>;
```

Comment:
This works, but it doubles the amount of memory that is being allocated.  A 20mb message would consume 40mb+ of memory because of both $messagetext and $payload containing the data.  I wonder if it makes more sense to strip out the carriage returns earlier in the process, when the queue file is written to the milterout queue.

Reply:
Here's a PR that should do the same thing to the outbound queue file itself as it is written.  https://github.com/MailScanner/v5/pull/315

Reply:
Thanks @shawniverson  I knew there'd be a better way.